src/crafting.cpp

```diff
@@ -605,24 +605,32 @@ int recipe::batch_time( int batch ) const
 
     float local_time = float( time ) / lighting_speed;
 
-    if( batch_rscale == 0.0 ) {
-        return local_time * batch;
-    }
-
     // NPCs around you should assist in batch production if they have the skills
     const auto helpers = g->u.get_crafting_helpers();
     int assistants = std::count_if( helpers.begin(), helpers.end(),
     [this]( const npc * np ) {
         return np->get_skill_level( skill_used ) >= difficulty;
     } );
 
+    // if recipe does not benefit from batching and we have no assistants, don't do unnecessary additional calculations
+    if( batch_rscale == 0.0 && assistants == 0 ) {
+        return local_time * batch;
+    }
+    
     double total_time = 0.0;
-    // At batch_rsize, incremental time increase is 99.5% of batch_rscale
-    double scale = batch_rsize / 6.0;
-    for( int x = 0; x < batch; x++ ) {
-        // scaled logistic function output
-        double logf = ( 2.0 / ( 1.0 + exp( -( ( double )x / scale ) ) ) ) - 1.0;
-        total_time += ( double )local_time * ( 1.0 - ( batch_rscale * logf ) );
+    // if recipe does not benefit from batching but we do have assistants, skip calculating the batching scale factor
+    if( batch_rscale == 0.0 ) {
+        total_time = ( double )local_time * batch;
```

Comment:
I didn't write most of that code, but Jenkins threw an error when I didn't explicitly cast local_time as a double on line 623.